File: src/openfl/events/TimerEvent.nim
import ./Event

type
  TimerEvent* = ref object of Event

const
  TIMER* = "timer"
  TIMER_COMPLETE* = "timerComplete"

proc newTimerEvent*(typeStr: string, bubbles: bool = false, cancelable: bool = false): TimerEvent =
  let self = TimerEvent()
  self.`type` = typeStr
  self.bubbles = bubbles
  self.cancelable = cancelable
  self.kindPreventDefaultFlag = false # Reusing for updateAfterEvent if needed, or add new field
  return self

method clone*(self: TimerEvent): Event =
  let event = newTimerEvent(self.`type`, self.bubbles, self.cancelable)
  event.target = self.target
  event.currentTarget = self.currentTarget
  event.eventPhase = self.eventPhase
  return event

method toString*(self: TimerEvent): string =
  "[TimerEvent type=" & self.`type` & " bubbles=" & $self.bubbles & " cancelable=" & $self.cancelable & "]"

proc updateAfterEvent*(self: TimerEvent) =
  # self.updateAfterEventFlag = true
  discard

File: kfl/src/main/kotlin/openfl/events/TimerEvent.kt
package openfl.events

class TimerEvent(
    type: String,
    bubbles: Boolean = false,
    cancelable: Boolean = false
) : Event(type, bubbles, cancelable) {

    companion object {
        const val TIMER = "timer"
        const val TIMER_COMPLETE = "timerComplete"
    }

    internal var updateAfterEventFlag: Boolean = false

    override fun clone(): TimerEvent {
        val event = TimerEvent(type, bubbles, cancelable)
        event.target = target
        event.currentTarget = currentTarget
        event.eventPhase = eventPhase
        return event
    }

    override fun toString(): String {
        return formatToString("TimerEvent", "type", "bubbles", "cancelable")
    }

    fun updateAfterEvent() {
        updateAfterEventFlag = true
    }
}
